File: python/4. winnerTeam.py
teams = [
        ['html', 'c#'],
        ['c#', 'python'],
        ['python', 'html'],
]

results = [0, 0, 1]

# Time Complexity O(N)
# Time Complexity O(K)


def winnerTeam(teams, results):
    bestTeam = ""
    score = {bestTeam: 0}
    WINNER_TEAM = 1

    for idx, team in enumerate(teams):
        result = results[idx]

        homeTeam, awayTeam = team

        wonTeam = homeTeam if result == WINNER_TEAM else awayTeam

        if wonTeam in score.keys():
            score[wonTeam] += 3
        else:
            score[wonTeam] = 3

        if(score[wonTeam] > score[bestTeam]):
            bestTeam = wonTeam

    return bestTeam


winnerTeam(teams, results)

File: js/4. winnerTeam.js
const teams = [
	['html', 'c#'],
	['c#', 'python'],
	['python', 'html'],
];

const results = [0, 0, 1];

// Time Complexity O(N)
// Time Complexity O(K)

function winnerTeam(teams, results) {
	const WINNER_TEAM = 1;

	let bestTeam = '';

	let score = { '': 0 };

	for (let [index, team] of teams.entries()) {
		const result = results[index];

		const [homeTeam, awayTeam] = team;

		let winnerTeam = result === WINNER_TEAM ? homeTeam : awayTeam;

		if (score[winnerTeam] !== undefined) {
			score[winnerTeam] = score[winnerTeam] + 3;
		} else {
			score[winnerTeam] = 3;
		}

		if (score[winnerTeam] > score[bestTeam]) bestTeam = winnerTeam;
	}

	return bestTeam;
}

winnerTeam(teams, results);
